### PR TITLE
Support .appup.src files

### PR DESCRIPTION
**Pull request by lrascao**
Similar to how .app.src are being handled, all files in directory
.src ending with .appup.src are processed, checked for
correctness and copied to ebin without the .src extension.

By way of using rebar_config:consult_file/1, .appup.src.script
files are automatically supported in exactly the same manner
as in .app.src.script.

### CODE REVIEW

File: src/rebar_otp_appup.erl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+info(help, compile) ->
+    info_help("Validate .appup.src file");
+info(help, clean) ->
+    info_help("Delete .appup file if generated from .appup.src").
+
+info_help(Description) ->
+    ?CONSOLE(
+       "~s.~n"
+       "~n"
+       "Valid rebar.config options:~n",
```
Comment (tuncer) [inline src/rebar_otp_appup.erl]: As there are no options, we should omit that bit completely as we do in `rebar_qc:info(help, clean)`.

File: src/rebar_otp_appup.erl [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+       [Description]).
+
+preprocess(SourceFile, TargetFile, _Config) ->
+    %% Perform basic validation on the appup file
+    %% ie. if a consult succeeds and basic appup structure exists
+    case rebar_config:consult_file(SourceFile) of
+        %% The .appup syntax is described in
+        %% http://erlang.org/doc/man/appup.html
```
Comment (tuncer) [inline src/rebar_otp_appup.erl]: Maybe terminate the sentence with a period?

File: src/rebar_otp_appup.erl [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+info_help(Description) ->
+    ?CONSOLE(
+       "~s.~n"
+       "~n"
+       "Valid rebar.config options:~n",
+       [Description]).
+
+preprocess(SourceFile, TargetFile, _Config) ->
+    %% Perform basic validation on the appup file
+    %% ie. if a consult succeeds and basic appup structure exists
```
Comment (tuncer) [inline src/rebar_otp_appup.erl]: `ie.` -> `i.e.` and maybe terminate the sentence with a period?

File: src/rebar_otp_appup.erl [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    case rebar_config:consult_file(SourceFile) of
+        %% The .appup syntax is described in
+        %% http://erlang.org/doc/man/appup.html
+        {ok, [{_Vsn, [_UpFromVsn], [_DownToVsn]} = AppUp]} ->
+            file:write_file(TargetFile,
+                            lists:flatten(io_lib:format("~p.",[AppUp]))),
```
Comment (tuncer) [inline src/rebar_otp_appup.erl]: Shouldn't we handle a failed `write_file`?

Comment (tuncer) [inline src/rebar_otp_appup.erl]: Missing space in `",[A`?

File: inttest/appup_src_script/src/app.appup.src.script [hunk 5]
```diff
@@ -0,0 +1,9 @@
+% decompose the appup
```
Comment (tuncer) [inline inttest/appup_src_script/src/app.appup.src.script]: For consistency: `%` -> `%%`

**Comment by tuncer** [comment]
Once the issues raised in f4f9b5a are resolved, +1. And perhaps the commit summary should be `Support .appup.src[.script] files` instead.

**Comment by runcom** [comment]
:+1:

**Pull request merged**